src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -40,7 +46,38 @@
      * София - 900</p>
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+        if (fromDate.isAfter(toDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder sb = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo + '\n');
+        ArrayList<String[]> dataToArray = new ArrayList<>();
+        for (String s : data) {
+            if (isDateInRange(s.substring(0,10),fromDate,toDate)) {
+                dataToArray.add(s.split(" "));
+            }
+        }
```

Comment:
let's remove this block. You can check the range inside two loops above.ArrayList of String arrays it's two dimension array, see checklist